webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/queries/NodeFactQueryProcessor.scala

```diff
@@ -146,9 +162,11 @@ class NodeFactQueryProcessor(
         _   <- FactQueryProcessorLoggerPure.Metrics.debug(s"Analyse query in ${t1 - t0} ms")
         res <- nodeFactRepo
                  .getAll()(QueryContext.todoQC, s)
-                 .flatMap(all =>
-                   ZIO.filter(all.values)(node => FactQueryProcessorLoggerPure.debug(m.debugString) *> processOne(m, node))
-                 )
+                 .flatMap(all => {
+                   ZIO
+                     .filterPar(all.values)(node => FactQueryProcessorLoggerPure.debug(m.debugString) *> processOne(m, node))
+                     .withParallelism(4)
```

Comment:
a little optimisation that can reclam ~1/3 of the (small) process time. Here, we are just checking for all nodes in turn, so its highly parallelisable. But on the other hand, it's also extremelly cpu-cache efficient to just match over and over the same things. So 4 is arbitrary, but it should not take too much of a toil on small CPU machine, and still get a boost on higher count ones, without splitting unecessarly a Chunk of at most 10k nodes on too many small bits.